ast.go

```diff
@@ -54,6 +54,35 @@ func (ast *AST) PrettyPrint() {
 	spew.Dump("%#v", ast)
 }
 
+func SubtreeString (left string, operator string, right string) string {
+	var compactOps = map[string]bool{
+		"=": true,
+		"<": true,
+		">": true,
+		"<=": true,
+		">=": true,
+		"!=": true,
+	}
+
+	// if the operator never needs space combine and return
+	if compactOps[operator] {
+		return fmt.Sprintf("%s%s%s", left, operator, right)
+	}
+
+	leftRune := []rune(left)
+	rightRune := []rune(right)
+
+	if string(leftRune[len(leftRune) - 1]) != ")" {
```

Comment:
If you're comparing the last `byte` of `left` you can do: `if left[len(left)-1] != ')'`. Notice the change of double quotes to a single quote. `")"` is a string, but `')'` is a byte. 

Also, when working with strings or bytes, you can take a look at the [strings pkg](https://pkg.go.dev/strings) and [bytes pkg](https://pkg.go.dev/bytes) to see if there's something already implemented.